tests/test_component_media.py

```diff
@@ -851,40 +840,46 @@ class Media:
 
 @djc_test
 class TestMediaRelativePath:
-    class ParentComponent(Component):
-        template: types.django_html = """
-            {% load component_tags %}
-            <div>
-                <h1>Parent content</h1>
-                {% component "variable_display" shadowing_variable='override' new_variable='unique_val' %}
-                {% endcomponent %}
-            </div>
-            <div>
-                {% slot 'content' %}
-                    <h2>Slot content</h2>
-                    {% component "variable_display" shadowing_variable='slot_default_override' new_variable='slot_default_unique' %}
+    def _gen_parent_component(self):
```

Comment:
In multiple places in test files I had to wrap top-level component definitions in functions to be called from within tests, so that the state is properly cleared up between tests.